javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,419 @@
+"use strict";
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named updateAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  const newStrArr = [];
+  arr.forEach((string) => {
+    newStrArr.push(callback(string));
```

Comment:
consider using "map" here. It can really tighten things up
```js
const updateAnimal = (arr, callback) => {
  return arr.map(animal => callback(animal));
}
```